src/autocompleteDump.ts

```diff
@@ -0,0 +1,207 @@
+        $$: XMLOrderedFunctionReturnPart[],
+        [name: string]: XMLFunctionReturnPart[],
+    }>,
+    parameters: Array<{
+        $$: XMLOrderedFunctionParameterPart[],
+        [name: string]: XMLFunctionParameterPart[],
+    }>,
+    description: string[],
+}
+
+interface XMLGroup {
+    $: {
+        name: string,
+    },
+    $$: [], // Shouldnt be used
+    Function?: XMLFunction[],
+    Properties?: XMLProperty[],
+}
+
+interface XMLTree {
+    StudioAutocomplete: {
+        LuaLibrary: XMLGroup[],
+        ItemStruct: XMLGroup[],
+        CoreLibrary: XMLGroup[]
+        ReservedWords: Array<{
+            $$: Array<{
+                $: {
+                    name: string,
+                },
+                "#name": string,
+            }>,
+            Keyword: Array<{
+                $: {
+                    name: string,
+                },
+            }>,
+        }>,
+    }
+}
+
+const formatFunction = (func: XMLFunction): AutocompleteFunction => {
+    const attributes = func.$
+    const parameters: AutocompleteParameter[] =
+        (func.parameters && func.parameters[0].$$) ? func.parameters[0].$$.map(paramObj => {
+            const type = paramObj["#name"]
+            const parameterAttributes = paramObj.$
+            return {
+                constraint: parameterAttributes.constraint,
+                name: parameterAttributes.name,
+                optional: parameterAttributes.optional === "true",
+                type,
+            }
+        }) : []
+
+    const returns = (func.returns && func.returns[0].$$) ? func.returns[0].$$.map(ret => {
+        const type = ret["#name"]
+        const returnAttributes = ret.$
+        return {
+            name: returnAttributes && returnAttributes.name,
+            type,
+        }
+    }) : []
+
+    return {
+        description: func.description && func.description[0],
+        name: attributes.name,
+        parameters,
+        returns,
+        static: attributes.static === "true",
+    }
+}
+
+const formatProperty = (property: XMLOrderedPropertyPart): AutocompleteProperty => {
+    const description = property._
+    const type = property["#name"]
+    const attributes = property.$
+    return {
+      description,
+      name: attributes.name,
+      static: attributes.static === "true",
+      type,
+    }
+}
+
+const formatGroup = (group: XMLGroup): AutocompleteGroup => {
+    return {
+      functions: group.Function ? group.Function.map(formatFunction) : [],
+      name: group.$.name,
+      properties: (group.Properties && group.Properties[0].$$) ? group.Properties[0].$$.map(formatProperty) : [],
+    }
+}
+
+const formatter = (tree: XMLTree): AutocompleteDump => {
+    const root = tree.StudioAutocomplete
+    const LuaLibrary = root.LuaLibrary
+    const ItemStruct = root.ItemStruct
+
+    const updatedLibrary = LuaLibrary.map(formatGroup)
+    const updatedItemStruct = ItemStruct.map(formatGroup)
+
+    return {
+      ItemStruct: updatedItemStruct,
```

Comment:
Not sure this indenting is right. There should be a VSC command to auto-fix this kind of stuff.